server/kv.go

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package main
+
+import (
+	"crypto/md5"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+)
+
+const (
+	keyKnownJIRAInstances  = "known_jira_instances"
+	keyTokenSecret         = "token_secret"
+	keyRSAKey              = "rsa_key"
+	keyCurrentJIRAInstance = "current_jira_instance"
+	prefixJIRAUserInfo     = "mm_j_" // + Mattermost user ID
+	prefixMattermostUserId = "j_mm_" // + JIRA username
+	prefixJIRAInstance     = "jira_instance_"
+)
+
+const prefixForInstance = true
+
+const (
+	JIRACloudType  = "cloud"
+	JIRAServerType = "server"
+)
+
+type JIRAInstance struct {
+	Key                         string
+	Type                        string
+	AtlassianSecurityContextRaw string
+	asc                         *AtlassianSecurityContext `json:"none"`
```

Comment:
Maybe write this out as `SecurityContext` it's creating some really cryptic calls like `ji.asc.`

Reply:
y, JIRAInstance need to be refactored to become the provider of JIRA clients, other methods that don't need the full p object. Will take care of it as I am adding the equivalent for JIRA server now.